src/main/java/frc/robot/subsystems/Indexer.java

```diff
@@ -86,103 +91,85 @@ public CommandBase reverseOutput() {
     // This command will make sure that the singulator has possesion of the ball
 
     public CommandBase pierrePossesion() {
-        return new FunctionalCommand(() -> {
-
+        CommandBase cmd = new FunctionalCommand(() -> {
         }, () -> {
-
             if (frontIntakeIR.getAsBoolean()) {
-
                 singulator.set(singulatorMotorSpeed);
             }
-            if (backIntakeIR.getAsBoolean()) {
+            if (backIntakeIR.getAsBoolean() && !topPierreIR.getAsBoolean()) {
                 pierreMotor.set(pierreIndexSpeed);
                 singulator.set(singulatorMotorSpeed);
             }
         }, (interrupted) -> {
-
             pierreMotor.set(0);
             singulator.set(0);
             SmartDashboard.putNumber("Ball Count", ballCounting);
-
         }, () -> {
-
-            return bottomPierreIR.getAsBoolean() || topPierreIR.getAsBoolean();
-
+            return (bottomPierreIR.getAsBoolean() && !backIntakeIR.getAsBoolean()) || topPierreIR.getAsBoolean();
         }, this);
-
+        cmd.setName("Pierre Possession");
+        return cmd;
     }
     // This command will make sure that pierre has possesion of the ball. It will be
     // at the bottom
 
     public CommandBase loadBallToTop() {
-        return new FunctionalCommand(() -> {
-
+        CommandBase cmd = new FunctionalCommand(() -> {
         }, () -> {
-
-            pierreMotor.set(pierreIndexSpeed);
-
+            if (!topPierreIR.getAsBoolean()) {
+                pierreMotor.set(pierreIndexSpeed);
+            }
         }, (interrupted) -> {
-
             pierreMotor.set(0);
-
         }, () -> {
-
             return topPierreIR.getAsBoolean();
-
         }, this);
-
+        cmd.setName("Load Ball to Top");
+        return cmd;
     }
 
     public CommandBase unLoadBall() {
-        return new FunctionalCommand(() -> {
-
+        CommandBase cmd = new FunctionalCommand(() -> {
         }, () -> {
-
             pierreMotor.set(pierreIndexSpeed);
-
         }, (interrupted) -> {
-
             pierreMotor.set(0);
             ballCounting--;
             SmartDashboard.putNumber("Ball Count", ballCounting);
-
         }, () -> {
-
             return !topPierreIR.getAsBoolean();
-
         }, this);
-
+        cmd.setName("Unload Ball");
+        return cmd;
     }
     // this will bring the ball to the shooter, it must already be at the top
 
+    public CommandBase shootAllBalls() {
+        CommandBase cmd = new SequentialCommandGroup(shootingBalls(), shootingBalls(), shootingBalls(), shootingBalls(),
```

Comment:
I think `CommandUtils.repeat(5, this::shootingBalls);` is the line you are looking for.

Reply:
ok will try it

Reply:
Type mismatch: cannot convert from Command to CommandBaseJava(16777233)

Reply:
Fixed